File: SRXLite/Controls/TextBoxCalendar.ascx.vb
Imports SRXLite.Modules

Namespace Web.Controls

    Partial Class Controls_TextBoxCalendar
        Inherits System.Web.UI.UserControl

        Private _selectedDate As Date
        Private _textBoxOnChange As String
        Private _textBoxValue As String

#Region " Properties "
        Public Property SelectedDate() As Date
            Get
                Return DateCheck(txtDate.Text, Nothing)
            End Get
            Set(ByVal value As Date)
                txtDate.Text = value.ToShortDateString
            End Set
        End Property

        Public ReadOnly Property TextBox() As TextBox
            Get
                Return txtDate
            End Get
        End Property

        Public Property TextBoxOnChange() As String
            Get
                Return _textBoxOnChange
            End Get
            Set(ByVal value As String)
                _textBoxOnChange = value
                txtDate.Attributes.Add("onchange", _textBoxOnChange)
            End Set
        End Property

        Public Property Value() As String
            Get
                Return txtDate.Text.Trim
            End Get
            Set(ByVal value As String)
                txtDate.Text = value
            End Set
        End Property
#End Region

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        End Sub

        Protected Sub txtDate_TextChanged(sender As Object, e As EventArgs)

        End Sub
    End Class

End Namespace

File: SRXLite.CS/Controls/TextBoxCalendar.ascx.cs
using System;
using System.Web.UI.WebControls;
using static SRXLite.Modules.Util;

namespace SRXLite.Web.Controls
{
    public partial class Controls_TextBoxCalendar : System.Web.UI.UserControl
    {
        private DateTime _selectedDate;
        private string _textBoxOnChange;
        private string _textBoxValue;

        #region Properties

        public DateTime SelectedDate
        {
            get { return DateCheck(txtDate.Text, DateTime.MinValue); }
            set { txtDate.Text = value.ToShortDateString(); }
        }

        public TextBox TextBox
        {
            get { return txtDate; }
        }

        public string TextBoxOnChange
        {
            get { return _textBoxOnChange; }
            set
            {
                _textBoxOnChange = value;
                txtDate.Attributes.Add("onchange", _textBoxOnChange);
            }
        }

        public string Value
        {
            get { return txtDate.Text.Trim(); }
            set { txtDate.Text = value; }
        }

        #endregion

        protected void Page_Load(object sender, EventArgs e)
        {
        }

        protected void txtDate_TextChanged(object sender, EventArgs e)
        {
        }
    }
}
